subprojects/docs/src/docs/userguide/javaLibraryPlugin.xml

```diff
@@ -0,0 +1,277 @@
+            then directly lead to a compile error, which would be solved either by removing the type from the public API, or promoting the dependency as an
+            API dependency instead.
+        </para>
+    </section>
+
+    <section id="sec:java_library_configurations_graph">
+        <title>The Java library plugin configurations</title>
+        <para>The following graph describes the configuration setup when the Java library plugin is in use. The configurations in <firstterm>grey</firstterm> are
+            deprecated and only found for backwards compatibility.
+        </para>
+        <imageobject>
+            <imagedata fileref="img/java-library-configurations.png"/>
+        </imageobject>
+        <para>If we exclude the deprecated configurations, the next schema shows what you should care about as a user:</para>
+        <imageobject>
+            <imagedata fileref="img/java-library-ignore-deprecated.png"/>
+        </imageobject>
+        <itemizedlist>
+            <listitem>The configurations in <firstterm>green</firstterm> are the ones a user should use to declare dependencies
+            </listitem>
+            <listitem>The configurations in <firstterm>pink</firstterm> are the ones used when a component compiles, or runs against the library
+            </listitem>
+            <listitem>The configurations in <firstterm>blue</firstterm> are internal to the component, for its own use
+            </listitem>
+            <listitem>The configurations in <firstterm>white</firstterm> are legacy configurations that can be used to declare dependencies too
+            </listitem>
+        </itemizedlist>
+        <para>The role of each configuration is described in the following tables:</para>
+        <table>
+            <title>Java library plugin - configurations used to declare dependencies</title>
+            <thead>
+                <tr>
+                    <td>Configuration name</td>
+                    <td>Role</td>
+                    <td>Can be consumed</td>
+                    <td>Can be resolved</td>
+                    <td>Description</td>
+                </tr>
+            </thead>
+            <tr>
+                <td>api</td>
+                <td>Declaring API dependencies</td>
+                <td>no</td>
+                <td>no</td>
+                <td>This is where you should declare dependencies which are transitively exported to consumers, for compile.</td>
+            </tr>
+            <tr>
+                <td>implementation</td>
+                <td>Declaring implementation dependencies</td>
+                <td>no</td>
+                <td>no</td>
+                <td>This is where you should declare dependencies which are purely internal and not meant to be exposed to consumers.</td>
+            </tr>
+            <tr>
+                <td>compileOnly</td>
```

Comment:
Maybe we should highlight in the table that `compileOnly` and `testCompileOnly` are (going to be) deprecated and that `implementation` should be used whenever possible?

E.g: `<td>compileOnly (*deprecated*)</td>`